src/lib/operations.ts

```diff
@@ -64,6 +66,114 @@ interface StdLibCallInfo {
   supportsTransform?: boolean
 }
 
+// Helper functions for argument extraction
+async function extractKclArgument(
+  operation: StdLibCallOp,
+  argName: string,
+  isArray = false
+): Promise<KclCommandValue | { error: string }> {
+  const arg = operation.labeledArgs?.[argName]
+  if (!arg?.sourceRange) {
+    return { error: `Missing or invalid ${argName} argument` }
+  }
+
+  const result = await stringToKclExpression(
+    codeManager.code.slice(arg.sourceRange[0], arg.sourceRange[1]),
+    isArray
+  )
+
+  if (err(result) || 'errors' in result) {
+    return { error: `Failed to parse ${argName} argument as KCL expression` }
+  }
+
+  return result
+}
+
+/**
+ * Extracts face selections for GDT annotations.
+ *
+ * Handles following types of face selections through direct tagging:
+ * - Segment faces: Tagged directly on segments, converted to wall artifacts
+ * - Cap faces: Tagged directly on sweeps using tagEnd/tagStart
+ * GDT uses direct tagging for explicit face references.
+ */
+function extractFaceSelections(facesArg: any): Selection[] | { error: string } {
```

Comment:
Wondering how this differs from an existing util function that we have for this, will look as well